From 695cd76e9683e10e17365dc1a1dc2d8b579f8e64 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Tue, 8 Oct 2024 06:48:13 +0000
Subject: [PATCH 02/43] LOCAL: test uart_ao_a_pins bias disable on Odroid C2
 and WeTek Hub

This resolves reports of non-booting C2 boards. The long term fix
is [0] which is merged for Linux 6.16 and will appear in u-boot
with a future upstream device-tree sync.

[0] https://patchwork.kernel.org/project/linux-amlogic/cover/20250329185855.854186-1-martin.blumenstingl@googlemail.com/

Suggested-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 arch/arm/dts/meson-gxbb-odroidc2-u-boot.dtsi  | 7 +++++++
 arch/arm/dts/meson-gxbb-wetek-hub-u-boot.dtsi | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/arch/arm/dts/meson-gxbb-odroidc2-u-boot.dtsi b/arch/arm/dts/meson-gxbb-odroidc2-u-boot.dtsi
index 1bd000eaa86..688c53c9d77 100644
--- a/arch/arm/dts/meson-gxbb-odroidc2-u-boot.dtsi
+++ b/arch/arm/dts/meson-gxbb-odroidc2-u-boot.dtsi
@@ -35,6 +35,13 @@
 	snps,reset-active-low;
 };
 
+&uart_ao_a_pins {
+	mux {
+		/delete-property/ bias-disable;
+		bias-pull-up;
+	};
+};
+
 &usb0 {
 	status = "disabled";
 };
diff --git a/arch/arm/dts/meson-gxbb-wetek-hub-u-boot.dtsi b/arch/arm/dts/meson-gxbb-wetek-hub-u-boot.dtsi
index cd2e9b5099e..baa704fb1b2 100644
--- a/arch/arm/dts/meson-gxbb-wetek-hub-u-boot.dtsi
+++ b/arch/arm/dts/meson-gxbb-wetek-hub-u-boot.dtsi
@@ -11,3 +11,10 @@
 	snps,reset-delays-us = <0 10000 1000000>;
 	snps,reset-active-low;
 };
+
+&uart_ao_a_pins {
+	mux {
+		/delete-property/ bias-disable;
+		bias-pull-up;
+	};
+};
-- 
2.43.0

